src/test/scala/mesosphere/marathon/core/election/impl/ElectionServiceBaseTest.scala

```diff
@@ -0,0 +1,269 @@
+package mesosphere.marathon.core.election.impl
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import akka.event.EventStream
+import com.codahale.metrics.MetricRegistry
+import mesosphere.chaos.http.HttpConf
+import mesosphere.marathon.core.base.ShutdownHooks
+import mesosphere.marathon.event.LocalLeadershipEvent
+import mesosphere.marathon.{ MarathonSpec, MarathonConf }
+import mesosphere.marathon.core.election.{ ElectionCandidate, ElectionService }
+import mesosphere.marathon.metrics.Metrics
+import mesosphere.marathon.test.MarathonActorSupport
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+import org.mockito.Mockito
+import org.rogach.scallop.ScallopOption
+import org.scalatest.{ GivenWhenThen, BeforeAndAfterAll, Matchers }
+
+import scala.concurrent.duration._
+
+class ElectionServiceBaseTest
+    extends MarathonActorSupport
+    with MarathonSpec
+    with GivenWhenThen
+    with BeforeAndAfterAll
+    with Matchers {
+
+  import ElectionServiceBase._
+
+  class Fixture {
+    val MaxActorStartupTime = 5000L
+    val OnElectedPrepareTimeout = 3 * 60 * 1000L
+
+    val config: MarathonConf = mockConfig
+    val httpConfig: HttpConf = mock[HttpConf]
+    val electionService: ElectionService = mock[ElectionService]
+    val events: EventStream = new EventStream()
+    val candidate: ElectionCandidate = mock[ElectionCandidate]
+    val metrics: Metrics = new Metrics(new MetricRegistry)
+    val backoff: Backoff = new ExponentialBackoff(0.01.seconds, 0.1.seconds)
+    val shutdownHooks: ShutdownHooks = mock[ShutdownHooks]
+
+    def mockConfig: MarathonConf = {
```

Comment:
I understand that you took this from already existing tests, but we're trying to move to a new cleaner style, and since this is a new test suite, there's no reason for adopting the old one.

There are now two helper methods to create configuration objects in tests: `MarathonTestHelper.makeConfig` and `MarathonTestHelper.defaultConfig`. You can see example usages here: https://github.com/mesosphere/marathon/blob/master/src/test/scala/mesosphere/marathon/MarathonConfTest.scala